mypy/subtypes.py

```diff
@@ -302,6 +331,237 @@ def visit_type_type(self, left: TypeType) -> bool:
+                # attributes with the corresponding return type. If this will produce
+                # many false negatives, then this could be prohibited for
+                # structural subtyping.
+                method = info.get_method(method_name)
+                if method and method.info.fullname() != 'builtins.object':
+                    getattr_type = find_node_type(method, itype, subtype)
+                    if isinstance(getattr_type, CallableType):
+                        return getattr_type.ret_type
+        if itype.type.fallback_to_any:
+            return AnyType(TypeOfAny.special_form)
+    return None
+
+
+def get_member_flags(name: str, info: TypeInfo) -> Set[int]:
+    """Detect whether a member 'name' is settable, whether it is an
+    instance or class variable, and whether it is class or static method.
+    """
+    method = info.get_method(name)
+    setattr_meth = info.get_method('__setattr__')
+    if method:
+        # this could be settable property
+        if method.is_property:
+            assert isinstance(method, OverloadedFuncDef)
+            dec = method.items[0]
+            assert isinstance(dec, Decorator)
+            if dec.var.is_settable_property or setattr_meth:
+                return {IS_SETTABLE}
+        return set()
+    node = info.get(name)
+    if not node:
+        if setattr_meth:
+            return {IS_SETTABLE}
+        return set()
+    v = node.node
+    if isinstance(v, Decorator):
+        if v.var.is_staticmethod or v.var.is_classmethod:
+            return {IS_CLASS_OR_STATIC}
+    # just a variable
+    if isinstance(v, Var):
+        flags = {IS_SETTABLE}
+        if v.is_classvar:
+            flags.add(IS_CLASSVAR)
+        return flags
+    return set()
+
+
+def find_node_type(node: Union[Var, FuncBase], itype: Instance, subtype: Type) -> Type:
+    """Find type of a variable or method 'node' (maybe also a decorated method).
+    Apply type arguments from 'itype', and bind 'self' to 'subtype'.
+    """
+    from mypy.checkmember import bind_self
+    if isinstance(node, FuncBase):
+        typ = function_type(node,
+                            fallback=Instance(itype.type.mro[-1], []))  # type: Optional[Type]
+    else:
+        typ = node.type
+    if typ is None:
+        return AnyType(TypeOfAny.from_error)
+    # We don't need to bind 'self' for static methods, since there is no 'self'.
+    if isinstance(node, FuncBase) or isinstance(typ, FunctionLike) and not node.is_staticmethod:
+        assert isinstance(typ, FunctionLike)
+        signature = bind_self(typ, subtype)
+        if node.is_property:
+            assert isinstance(signature, CallableType)
+            typ = signature.ret_type
+        else:
+            typ = signature
+    itype = map_instance_to_supertype(itype, node.info)
+    typ = expand_type_by_instance(typ, itype)
+    return typ
+
+
+def get_missing_members(left: Instance, right: Instance) -> List[str]:
```

Comment:
This doesn't seem like the right module for this function. I'd suggest moving this to `messages.py`. Also maybe rename this to `get_missing_protocol_members`.